core/common/src/ITwinError.ts

```diff
@@ -0,0 +1,92 @@
+import { BentleyError, LoggingMetaData } from "@itwin/core-bentley";
+import { LockState } from "./IModelError";
+
+/**
+ * Detailed information about a particular object Lock that is causing the Lock update conflict.
+ * An example of a lock update conflict would be attempting to use [LockControl.acquireLocks]($backend) on an object that is already locked by another Briefcase.
+ * @beta
+ */
+export interface InUseLock {
+/** Id of the object that is causing conflict. */
+  objectId: string;
+  /**
+   * The level of conflicting lock. Possible values are {@link LockState.Shared}, {@link LockState.Exclusive}.
+   * See {@link LockState}.
+   */
+  state: LockState;
+  /** An array of Briefcase ids that hold this lock. */
+  briefcaseIds: number[];
+}
+
+/**
+ * An interface used to describe an error. This error interface should be extended when needing to throw errors with extra properties defined on them. See [[InUseLocksError]] for an example.
+ * Also see [[ITwinError.throwInUseLocksError]] and [[ITwinError.isInUseLocksError]] for examples of how to throw and check that an error is of type InUseLocksError.
+ * @beta
+ */
+export interface ITwinError {
+  /** namespace for the error */
+  namespace: string;
+  /** unique key for error, within namespace */
+  errorKey: string;
+  /** explanation of what went wrong. */
+  message: string;
+  /** stack trace of the error. */
+  stack?: string;
+  /** metadata about the exception. */
+  metadata?: LoggingMetaData;
+}
+
+/**
+ * An error raised when there is a lock conflict detected.
+ * Typically this error would be thrown by [LockControl.acquireLocks]($backend) when you are requesting a lock on an element that is already held by another briefcase.
+ * @beta
+*/
+export interface InUseLocksError extends ITwinError {
+  namespace: "IModelAccess";
```

Comment:
What level of granularity on namespace is really wanted/useful? Seems like you could use a single namespace for all of itwinjs-core.

Reply:
Yeah thats a good point. I switched to "itwinjs-core". Now it feels a little weird that the case of the errorKey and the case of the namespace differ, but "ITwinJsCore" feels wrong to me. Not sure if you have an opinion on that.

Reply:
I don't feel strongly about it, but if you want to establish consistency consider doing `"in-use-locks"`. People shouldn't be comparing namespace or key themselves anyway, they should be calling a type assertion function that does the comparison for them.

Reply:
Switched to "in-use-locks", thanks